### PR TITLE
[RESTEASY-2131] Implements use of MicroProfile Config.

### PR DESCRIPTION
**Pull request by ronsigal**

### CODE REVIEW

File: resteasy-core/src/main/java/org/jboss/resteasy/microprofile/config/FilterConfigSource.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      }
+      return config.getInitParameter(propertyName);
+   }
+
+   @Override
+   public String getName() {
+      ServletContext servletContext = ResteasyContext.getContextData(ServletContext.class);
+      FilterConfig filterConfig = ResteasyContext.getContextData(FilterConfig.class);
+      ServletConfig servletConfig = ResteasyContext.getContextData(ServletConfig.class);
+      return (servletContext != null ? servletContext.getServletContextName() : null) +
```
Comment (asoldano) [inline resteasy-core/src/main/java/org/jboss/resteasy/microprofile/config/FilterConfigSource.java]: I'm wondering whether this is something that should be computed once, stored and then reused. The reason being basically 1) stability of the result (is it possible that the name changes between multiple invocations?) 2) performances (there're multiple string concat involved in computing the name). WDYT?

File: resteasy-core/src/main/java/org/jboss/resteasy/microprofile/config/ServletConfigSource.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      }
+      return config.getInitParameter(propertyName);
+   }
+
+   @Override
+   public String getName() {
+      ServletContext servletContext = ResteasyContext.getContextData(ServletContext.class);
+      FilterConfig filterConfig = ResteasyContext.getContextData(FilterConfig.class);
+      ServletConfig servletConfig = ResteasyContext.getContextData(ServletConfig.class);
+      return (servletContext != null ? servletContext.getServletContextName() : null) +
```
Comment (asoldano) [inline resteasy-core/src/main/java/org/jboss/resteasy/microprofile/config/ServletConfigSource.java]: Same comment as above about name computation

File: resteasy-core/src/main/java/org/jboss/resteasy/microprofile/config/ServletContextConfigSource.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      }
+      return context.getInitParameter(propertyName);
+   }
+
+   @Override
+   public String getName() {
+      ServletContext servletContext = ResteasyContext.getContextData(ServletContext.class);
+      FilterConfig filterConfig = ResteasyContext.getContextData(FilterConfig.class);
+      ServletConfig servletConfig = ResteasyContext.getContextData(ServletConfig.class);
+      return (servletContext != null ? servletContext.getServletContextName() : null) +
```
Comment (asoldano) [inline resteasy-core/src/main/java/org/jboss/resteasy/microprofile/config/ServletContextConfigSource.java]: Same comment as above about name computation

File: resteasy-core/src/main/java/org/jboss/resteasy/microprofile/config/FilterConfigSource.java (hunk 1 above)
Comment (ronsigal) [reply resteasy-core/src/main/java/org/jboss/resteasy/microprofile/config/FilterConfigSource.java]: Fair enough. I was thinking it was something that wouldn't get called a lot, but you never know. Also, I simplified the names. For example, in RESTEasy, starting with a Servlet and starting with a Filter are mutually exclusive.

File: resteasy-core/src/main/java/org/jboss/resteasy/microprofile/config/FilterConfigSource.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+         return null;
+      }
+      return config.getInitParameter(propertyName);
+   }
+
+   @Override
+   public String getName() {
+      if (name == null) {
+         synchronized(this) {
+            if (name == null) {
```
Comment (asoldano) [inline resteasy-core/src/main/java/org/jboss/resteasy/microprofile/config/FilterConfigSource.java]: @ronsigal , shouldn't the "name" field be volatile for this to work properly?

Comment (ronsigal) [reply resteasy-core/src/main/java/org/jboss/resteasy/microprofile/config/FilterConfigSource.java]: D'oh. I knew there was something bothering me about that. Fixed. Thanks.

**Pull request merged**